app/src/main/kotlin/compiler/intermediate_form/ControlFlowGraphBuilder.kt

```diff
@@ -43,7 +50,15 @@ class ControlFlowGraphBuilder(@JvmField var entryTreeRoot: IFTNode? = null) {
     }
 
     fun addLinksFromAllFinalRoots(linkType: CFGLinkType, to: IFTNode) {
-        finalTreeRoots.forEach { addLink(Pair(it, linkType), to) }
+        finalTreeRoots.forEach {
+            if (it.second == CFGLinkType.UNCONDITIONAL)
+                addLink(it.copy(second = linkType), to)
```

Comment:
I think that `addLink(it.first to linkType, to)` is more readable

Reply:
Changed to `Pair(it.first, linkType)`, because the two meanings of `to` may be confusing.